pkg/defaults/hostname_generator.go

```diff
@@ -98,12 +98,15 @@ func EnsureHostnameGeneratorExists(ctx context.Context, resManager core_manager.
 			},
 			Template: `{{ .DisplayName }}.extsvc.mesh.local`,
 		}
-		if err := ensureHostnameGeneratorExists(ctx, resManager, log, "synced-mesh-external-service", namespace, spec); err != nil {
+		if err := ensureHostnameGeneratorExists(ctx, resManager, log, "synced-mesh-external-service", namespace, nil, spec); err != nil {
 			return err
 		}
 	}
 
 	if cfg.Mode == core.Zone {
+		labels := map[string]string{
+			mesh_proto.ResourceOriginLabel: string(mesh_proto.ZoneResourceOrigin),
```

Comment:
@Automaat don't we already set this automatically?

Reply:
yes, but on api level :/ See https://github.com/kumahq/kuma/issues/11061

Reply:
I see, thanks for creating an issue